playbooks/roles/ocp-lso/tasks/main.yaml

```diff
@@ -0,0 +1,236 @@
+      when:
+        - lso_namespace is defined  # Ensure lso_namespace is defined
+    - name: Verify if local-storage-operator is present in the CSV output
+      debug:
+        msg: "The LocalStorage Operator CSV is available."
+      when: "'local-storage-operator' in csv_status.stdout"
+
+    # Create LocalVolumeSet YAML content for local block volumes
+    - name: Create LocalVolumeSet YAML for local block volumes
+      kubernetes.core.k8s:
+        state: present
+        definition:
+          kind: LocalVolumeSet
+          apiVersion: local.storage.openshift.io/v1alpha1
+          metadata:
+            name: localblock
+            namespace: "{{ lso_namespace }}"
+          spec:
+            deviceInclusionSpec:
+              deviceMechanicalProperties:
+                - Rotational
+                - NonRotational
+              deviceTypes:
+                - disk
+                - mpath
+              maxSize: 500Gi
+              minSize: 500Gi
+            maxDeviceCount: 10
+            nodeSelector:
+              nodeSelectorTerms:
+                - matchExpressions:
+                    - key: kubernetes.io/hostname
+                      operator: In
+                      values:
+                        - worker-0
+                        - worker-1
+                        - worker-2
+            storageClassName: localblock
+            volumeMode: Block
+
+    # Wait for LocalVolumeSet to be created
+    - name: Wait for LocalVolumeSet to be created
+      command: oc get localvolumeset localblock -n "{{ lso_namespace }}"
+      register: lvs_status
+      retries: 5
+      delay: 10
+      until: lvs_status.rc == 0
+      failed_when: lvs_status.rc != 0  # Fail if LocalVolumeSet is not created within 5 retries
+
+    - name: Verify creation of diskmanager pods
+      command: oc get pods -n "{{ lso_namespace }}" --selector app=diskmaker-manager
+      register: check_pods
+      failed_when: check_pods.rc != 0
+
+    - name: Verify the creation of StorageClass
+      command: oc get sc localblock
+      register: check_sc
+      failed_when: check_sc.rc != 0
+
+    - name: Wait for persistent volumes to be created
+      pause:
+        minutes: 1
+
+    - name: Verify the status of Persistent Volumes
+      command: oc get pv --selector storage.openshift.com/owner-kind=LocalVolumeSet
+      register: check_pv
+      failed_when: check_pv.rc != 0
+
+    - name: Delete existing PersistentVolumeClaim and Pod (if they exist)
+      kubernetes.core.k8s:
+        state: absent
+        definition:
+          - kind: PersistentVolumeClaim
+            apiVersion: v1
+            metadata:
+              name: test-local-claim
+          - apiVersion: v1
+            kind: Pod
+            metadata:
+              name: busybox-lso
+              namespace: default
+      register: delete_pvc
+      failed_when: false  # Do not fail if delete command is not successful (e.g., resource doesn't exist)
```

Comment:
Older deployments should be deleted first before deploying a newer one. Please remove `failed_when`

Reply:
removed